src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/posthearingreview/PostHearingReviewSubmittedHandler.java

```diff
@@ -56,38 +51,45 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
-        PostHearingReviewType typeSelected = postHearing.getReviewType();
-        log.info("Review Post Hearing App: handling actionPostHearing {} for case {}", typeSelected,  caseId);
+        Long caseId = Long.valueOf(caseData.getCcdCaseId());
 
-        CcdCallbackMap callbackMap = getCcdCallbackMap(postHearing, typeSelected);
+        log.info("Review Post Hearing App: handling actionPostHearing {} for case {}", typeSelected, caseId);
 
-        if (isNull(callbackMap)) {
+        if (isNull(typeSelected)) {
             response.addError(String.format("Invalid Action Post Hearing Application Type Selected %s or action "
-                    + "selected as callback is null",
-                typeSelected));
+                            + "selected as callback is null",
+                    typeSelected));
             return response;
         }
 
-        boolean isSetAsideRefusedSor = isSetAsideRefusedSor(postHearing);
+        updateCcdCaseService.updateCaseV2Conditional(
+                caseId,
+                EventType.SOR_REQUEST.getCcdType(),
+                idamService.getIdamTokens(),
+                sscsCaseDetails -> {
+                    var sscsCaseData = sscsCaseDetails.getData();
+                    var postHearing = sscsCaseData.getPostHearing();
 
-        SscsUtil.clearPostHearingFields(caseData, isPostHearingsEnabled);
+                    CcdCallbackMap callbackMap = getCcdCallbackMap(postHearing, typeSelected);
 
-        caseData = ccdCallbackMapService.handleCcdCallbackMap(callbackMap, caseData);
+                    boolean isSetAsideRefusedSor = isSetAsideRefusedSor(postHearing);
 
-        if (isSetAsideRefusedSor) {
-            handleSetAsideRefusedSor(caseData);
-        }
+                    SscsUtil.clearPostHearingFields(sscsCaseData, isPostHearingsEnabled);
 
-        return new PreSubmitCallbackResponse<>(caseData);
-    }
+                    ccdCallbackMapService.handleCcdCallbackMap(callbackMap, sscsCaseData);
+
+                    if (isSetAsideRefusedSor) {
+                        log.info("Review Post Hearing App - updateCaseV2Conditional triggered for case {} and type {} ", caseId, typeSelected);
+                        return new ConditionalUpdateResult("Send to hearing Judge for statement of reasons", "", true);
+                    }
+                    log.info("Review Post Hearing App - updateCaseV2Conditional not triggered for case {} and type {} ", caseId, typeSelected);
+                    return new ConditionalUpdateResult("Send to hearing Judge for statement of reasons", "", false);
+                }
+        );
 
-    private void handleSetAsideRefusedSor(SscsCaseData caseData) {
-        ccdService.updateCase(caseData, Long.valueOf(caseData.getCcdCaseId()),
-            EventType.SOR_REQUEST.getCcdType(), "Send to hearing Judge for statement of reasons", "",
-            idamService.getIdamTokens());
+        return new PreSubmitCallbackResponse<>(caseData);
```

Comment:
Should this set the case data returned from updateCcdCaseService.updateCaseV2Conditional if optional is not empty?